### PR TITLE
enable coverage in a github action

### LINKED ISSUE #2321: run coverage in github actions
See: #2315.

### PR DESCRIPTION
**Pull request by pq**
Fixes #2321.

image

🎉 🎉 🎉 

/cc @devoncarew @bwilkerson

### CODE REVIEW

**Comment by pq** [comment]
image

https://github.com/dart-lang/linter/pull/2432/checks?check_run_id=1770089764

@athomas:  could you advise?  What do I need to do to get my pub path properly  set up?

**Comment by pq** [comment]
(Update: fixed!)

**Comment by pq** [comment]
Interesting to see the travis check stalled:

image

Sure will be nice to move on!

**Review by devoncarew — APPROVED** [review]
You might want to just remove the travis.yaml file as well.

File: tool/travis.sh [hunk 1]
```diff
@@ -52,48 +52,36 @@ else
...
-
-    echo "Running coveralls..."
-    gem install coveralls-lcov
-    coveralls-lcov var/lcov.info
+  # Run the tests setup for coverage reporting.
+  dart --disable-service-auth-codes \
+    --disable-analytics \
+    --enable-vm-service=$OBS_PORT \
+    --pause-isolates-on-exit \
+    test/all.dart &
```
Comment (kevmoo) [inline tool/travis.sh:62]: Now we just need to figure out how to do this IN the "real" test run. CC @grouma

Comment (grouma) [reply tool/travis.sh:62]: `package:test` supports a `--coverage` argument.

**Pull request merged**